app/src/main/java/co/develhope/chooseyouowncocktail_g2/adapter/DrinkCardAdapter.kt

```diff
@@ -1,96 +1,117 @@
+            binding.drinkName.text = drink.name
 
+            binding.drinkShortDescription.text = drink.shortDescription
+            showPreferiteRecycleView(drink)
 
+            binding.drinkImage.setImageByUrl(
+                "https://www.thecocktaildb.com/images/media/drink/metwgh1606770327.jpg",
+                60,
+                60
+            )
 
+            binding.buttonGoToDetail.setOnClickListener {
+                action(DrinkAction.GotoDetail(drink.id))
+            }
 
+            binding.drinkFavourite.setOnClickListener {
+                drink.favourite = !drink.favourite
+                showPreferiteRecycleView(drink)
+                action(DrinkAction.SetPref(drink, drink.favourite))
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
-        val binding = DrinkCardBinding.inflate(LayoutInflater.from(parent.context), parent, false)
-        context= parent.getContext()
+                if (drink.favourite)
+                    preferDrink(position, 0)
+                else
+                    unPreferDrink(position)
 
+            }
+        }
 
+        private fun showPreferiteRecycleView(drink:Drink) {
+            if (drink.favourite) {
+                binding.drinkFavourite.setBackgroundResource(R.drawable.ic_fav_on)
 
-        return ViewHolder(binding)
+            } else {
+                binding.drinkFavourite.setBackgroundResource(R.drawable.ic_fav_off)
+            }
+        }
     }
 
-    override fun onBindViewHolder(holder: ViewHolder, position: Int,) {
+    override fun onCreateViewHolder(parent:ViewGroup, viewType:Int):ViewHolder{
+        val binding = DrinkCardBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        context = parent.context
 
-        with(receiver = holder) {
-            with(beerListForAdapter[position]) {
-                binding.drinkName.text = this.name
+        return ViewHolder(binding)
+    }
 
-                binding.drinkShortDescription.text = this.shortDescription
+    override fun onBindViewHolder(holder:ViewHolder, position:Int) {
+        holder.bind(drinkListForAdapter[position], position)
+    }
 
-                binding.drinkImage.setImageByUrl(
-                    "https://www.thecocktaildb.com/images/media/drink/metwgh1606770327.jpg",
-                    60,
-                    60
-                )
+    private fun preferDrink(toRemove:Int, toInsert:Int) {
 
+        val drink = (drinkListForAdapter as ArrayList<Drink>).removeAt(toRemove)
+        notifyItemRemoved(toRemove)
+        notifyItemChanged(toRemove)
 
-                binding.buttonGoToDetail.setOnClickListener {
-                    action(DrinkAction.GotoDetail(this.id))
-                }
-                // context.resources.getIdentifier("ic_fav_on", "drawable", context.packageName)
+        (drinkListForAdapter as ArrayList<Drink>).add(toInsert, drink)
+        notifyItemInserted(toInsert)
```

Comment:
anche questae azioni di modifica della lista non vanno effettuate nell'adapter ma nle fragment/activity che lo usa a che notifica le modifiche, anche perchè se stai moficiando la lista in qualche maniera probabilmente i valori cambiati vanno salvati e aggiornati nella originale